lib/defines.h

```diff
@@ -335,6 +335,13 @@ extern char *strerror ();
 # endif
 #endif
 
+/* [[gnu::malloc(deallocator)]] */
+#if defined(__GNUC__) && !defined(__clang__)
```

Comment:
Hmm, didn't know this is so new.

If it is so new, I'll go directly for the ISO C attribute syntax `[[]]`, I think, since I think they are already implemented in GCC 11.  I don't have a gcc-11 instance, so if you can confirm, it would be great.